### PR TITLE
chore: refactor utils

### PR DESCRIPTION
**Pull request by daomry**

### CODE REVIEW

File: packages/engineer/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@wixc3/engine-runtime-node": "^29.0.1",
     "@wixc3/engine-scripts": "^29.0.1",
     "commander": "^9.4.0",
+    "lodash": "^4.17.21",
```
Comment (AviVahl) [inline packages/engineer/package.json:27]: we decided to not include `lodash` in our packages.
it is unmaintained, and when it had security vulnerabilities they ignored PRs for months.
a lot of its functionality can be achieved via other means (new native syntax, or other libraries)

Comment (daomry) [reply packages/engineer/package.json:27]: good to know. is there a similar lib in use in our code?

Comment (AviVahl) [reply packages/engineer/package.json:27]: not a single library that contains many utils.
we got away we native syntax for most things

_shallow_ defaults, for example, is as simple as `{...a, ...b}`, while `unique` is just `Array.from(new Set(items))`, etc.

Comment (daomry) [reply packages/engineer/package.json:27]: great,  but I'm using defaultsDeep, which I need to implement and test... underscore.js?

Comment (daomry) [reply packages/engineer/package.json]: removed lodash dependency, now depends on https://github.com/wixplosives/core3-utils/pull/138

**Pull request closed without merging**